frontend/src/pages/ActivityReport/Pages/nextSteps.js

```diff
@@ -1,21 +1,193 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
 import { Helmet } from 'react-helmet';
+import { useFormContext, useWatch } from 'react-hook-form';
+import {
+  Fieldset, TextInput, Button,
+} from '@trussworks/react-uswds';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faPlusCircle } from '@fortawesome/free-solid-svg-icons';
+import ContextMenu from '../../../components/ContextMenu';
+import FormItem from '../../../components/FormItem';
 import ReviewPage from './Review/ReviewPage';
 
+const NoteEntry = ({
+  onEntry, onCancel, name, isRequired = false, defaultValue = '',
+}) => {
+  const [input, updateInput] = useState(defaultValue);
+
+  const onSubmit = (event) => {
+    event.preventDefault();
+    updateInput('');
+    onEntry(input.trim());
+  };
+
+  const onUpdate = (e) => {
+    e.preventDefault();
+    updateInput(e.target.value);
+  };
+
+  return (
+    <FormItem
+      required
+      fieldSetWrapper
+      name={name}
+      label="What have you agreed to do next?"
+    >
+      <TextInput name={name} onChange={onUpdate} data-testid={`${name}-input`} />
+      <Button outline disabled={!(input && input.trim())} onClick={onSubmit} data-testid={`${name}-button`}>Save Next Step</Button>
+      {!isRequired && <Button secondary onClick={onCancel} data-testid={`${name}-cancel-button`}>Cancel</Button>}
+    </FormItem>
+  );
+};
+
+NoteEntry.propTypes = {
+  onEntry: PropTypes.func.isRequired,
+  onCancel: PropTypes.func.isRequired,
+  name: PropTypes.string.isRequired,
+  defaultValue: PropTypes.string,
+  isRequired: PropTypes.bool,
+};
+
+NoteEntry.defaultProps = {
+  isRequired: false,
+  defaultValue: '',
+};
+
+const NoteEntries = ({ name, humanName }) => {
+  const { register, control, setValue } = useFormContext();
+  const notes = useWatch({ name, control });
+
+  const [showPrompt, updateShowPrompt] = useState(false);
+  const [targetIndex, updateTargetIndex] = useState(-1);
+
+  useEffect(() => {
+    register({ name }, { validate: (allNotes) => allNotes.length !== 0 });
+  });
+
+  const onEntry = (note, index, noteId) => {
+    const newNotes = [...notes];
```

Comment:
There is an issue I'm seeing with this page being erroneously marked as `Completed` or `In Progress`. If the page is `Completed` upon opening the page the state will always get set to `Completed` when I browse away, same with `In Progress`. Since the field is manually registered against `react-hook-form` the validate function is not being ran after the first render. There are two ways to handle it as I see:

1) pull `trigger` out of `useFormContext()` and call `trigger(name)` each time an entry is added/updated/deleted
2) Implement a `isPageComplete` method on the default export that checks if both notes have entries (see the goals & objectives page for an example)